daemon/cmd/app.go

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/cilium/cilium/api/v1/server"
+)
+
+const (
+	startTimeout = 5 * time.Minute
+	stopTimeout  = 5 * time.Minute
```

Comment:
If you grep around for `time` in `pkg/defaults/defaults.go`, you'll see some other similar startup timeouts, which I think are all configurable. This sort-of fits with them, but for instance if there are a lot of k8s objects to sync then it may make sense to bump this timeout... Maybe we can add an option for each of these to allow users to override the timeouts just in case?

Reply:
Currently there's no start hooks so the timeout does nothing. Perhaps the right thing to do here is just use that one context we have in daemon for both start and stop and then revisit when start and stop actually becomes meaningful?